mkosi/config.py

```diff
@@ -948,11 +946,11 @@ def config_match_enum(match: str, value: StrEnum) -> bool:
 def config_make_list_parser(
     *,
     delimiter: Optional[str] = None,
-    parse: Callable[[str], Any] = str,
+    parse: Callable[[str], T] = str,  # type: ignore # see mypy#3737
```

Comment:
I dislike type ignores and an explicit `cast(Callable[[str], T], str)` would work around this, but in this case this is probably not worth it here, but let's not link to the issue but make this
```suggestion
    parse: Callable[[str], T] = str,  # type: ignore # TODO: use PEP696 default type parameter once we are on 3.12
```
same in the other places.

Reply:
Sadly PEP 696 does not solve our problem, the issue/pep name is a bit confusing. The PEP is about global defaults for the TypeVar whereas what we need is inference of T based on default values for parameters whose annotation contains T

Reply:
Mhm, but shouldn't this work here? e.g. defining `ParserFunc` as alias for `Callable[[str], T]` with a default of `Callable[[str], str]` (that would still break, because `str` is seen as type and not as a function, so we'd need an `idparser = lambda s: s`, haven't checked this yet, though). I haven't read the full issue, but from the most recent posts I got the impression it would be closed due to PEP696.

Reply:
The issue was reopened because it is not sufficiently solved by PEP696 and it's my understanding that our use case if one of those not solved by the PEP